mtp_noms_ops/apps/security/forms.py

```diff
@@ -189,6 +200,100 @@ def get_api_endpoint(self):
         return self.client.credits.prisoners
 
 
+class AmountPattern(enum.Enum):
+    not_integral = _('Non-integer amount')
+    not_multiple_5 = _('Not a multiple of £5')
+    not_multiple_10 = _('Not a multiple of £10')
+    gte_100 = _('£100 or more')
+
+    @classmethod
+    def get_choices(cls):
+        return [(choice.name, choice.value) for choice in cls]
+
+    @classmethod
+    def get_filters(cls, amount_pattern):
+        filters = {}
+        try:
+            amount_pattern = cls[amount_pattern]
+            if amount_pattern == cls.not_integral:
+                filters['exclude_amount__endswith'] = '00'
+            elif amount_pattern == cls.not_multiple_5:
+                filters['exclude_amount__regex'] = '(500|000)$'
+            elif amount_pattern == cls.not_multiple_10:
+                filters['exclude_amount__endswith'] = '000'
+            elif amount_pattern == cls.gte_100:
+                filters['amount__gte'] = '10000'
+        except KeyError:
+            pass
+        return filters
+
+
 class CreditsForm(SecurityForm):
+    ordering = forms.ChoiceField(label=_('Sort by'), required=False,
+                                 initial='-received_at',
+                                 choices=[
+                                     ('-received_at', _('Received date')),
+                                     ('-amount', _('Amount')),
+                                     ('prisoner_number', _('Prisoner number')),
+                                     ('prisoner_name', _('Prisoner name')),
+                                 ])
+
+    amount = forms.CharField(label=_('Amount (exact)'), validators=[validate_amount], required=False)
+    amount_pattern = forms.ChoiceField(label=_('Amount pattern'), choices=AmountPattern.get_choices(), required=False)
+    amount_pence = forms.IntegerField(label=_('Amount pence part'), min_value=0, max_value=99, required=False)
+
+    prisoner_number = forms.CharField(label=_('Prisoner number'), validators=[validate_prisoner_number], required=False)
+    prisoner_name = forms.CharField(label=_('Prisoner name'), required=False)
+    prison = forms.ChoiceField(label=_('Prison'), required=False, choices=[])
+    prison_region = forms.ChoiceField(label=_('Prison region'), required=False, choices=[])
+    prison_gender = forms.ChoiceField(label=_('Prisoner gender'), required=False,
+                                      choices=(('m', _('Male')), ('f', _('Female'))))
```

Comment:
Relatively minor point of consistency, but any particular reason the choices are a tuple here but a list elsewhere?